public/app/features/explore/RichHistory/RichHistoryCard.tsx

```diff
@@ -147,24 +146,33 @@ export function RichHistoryCard(props: Props) {
   } = props;
   const [activeUpdateComment, setActiveUpdateComment] = useState(false);
   const [comment, setComment] = useState<string | undefined>(query.comment);
+  const [queryDsInstance, setQueryDsInstance] = useState<DataSourceApi | undefined>(undefined);
+
+  useEffect(() => {
+    getQueryDsInstance();
+  }, []);
```

Comment:
I think the deps should be `[query.datasourceName]` to run if query and datasource changes. If we do add keys to the cards in the list then changing the queries will just update the props and not unmount/mount new card.

Reply:
updated in https://github.com/grafana/grafana/pull/23627/commits/597b0a497c8a475ae9e872ea397f7dd8314bfdef